### DISCUSSION TITLE
API: Add Support for Simple Asymmetric Cryptography, Not Just KEM

### DISCUSSION THREAD

**Discussion by emendir** [open-quantum-safe/liboqs#1622] [Ideas]
As I understand it, liboqs provides two main sets of features:
- signing and signature verification
- key encapsulation mechanisms for hybrid encryption (compounding asymmetric and symmetric encryption)

As a programmer (who is not a cryptography expert), I wonder, Where is the functionality for simple asymmetric encryption?
It's great to have classes for the more complex hybrid encryption ready-to-go, but it's quite a let-down that the basic, fundamental underlying asymmetric encryption functionality implementing the quantum-resistant algoritms, which is after all, what this library is about, is unavailable in the APIs.

I know most applications use hybrid encryption instead of pure asymmetric encryption, but there are plenty of use-cases for pure asymmetric encryption.

Furthermore, for educational purposes, I think it is vital that users are given access to the fundamental mechanisms underlying higher-level complex constructs such as KEMs.

Adding access to the basic quatum-resistant asymmetric encyrption and decryption functions makes the API more wholesome and complete.
I was quite surprised and disappointed when looking at this cryptography library to see that it has no `encrypt` and `decrypt` functions in its API!

Am I missing a point here?
What do others think about this issue, I'm not exotic in my API-expectations and ideals, am I?!

**Comment by baentsch** [comment]
See [related discussion](https://github.com/orgs/open-quantum-safe/discussions/1464).

**Comment by emendir** [comment]
You're right.
Thanks for the tip.
It seems I didn't look hard enough for similar issues.
Sorry about that!
